### PR TITLE
[Dynamo] Fix llvm target for meta schedule & add torch to tvm ndarray helper func

### LINKED ISSUE #90213: [Dynamo] TVM's meta schedule throwing TVMError: Target does not have attribute "num-cores", physical core number must be defined!
### 🐛 Describe the bug

As in the title, the LLVM target of TVM in meta schedule needs to specify a `num-cores` number, as opposed to how it is defined in auto scheduler and autotvm.

### Versions

latest pytorch main

### PR DESCRIPTION
**Pull request by shingjan**
Fixes #90213. Also a torch.tensor to tvm.nd.array helper function is added to avoid data copy with dlpack.

@jansel @Chillee 

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

### CODE REVIEW

File: torch/_dynamo/optimizations/backends.py [hunk 1]
```diff
@@ -680,6 +684,14 @@ def to_torch_tensor(nd_tensor):
                 return torch.from_numpy(nd_tensor.numpy())
             return torch.utils.dlpack.from_dlpack(nd_tensor.to_dlpack())
 
+        def to_tvm_tensor(torch_tensor):
+            """A helper function to transfer a torch.tensor to NDArray."""
+            if torch_tensor.dtype == torch.bool:
+                # same reason as abpve, fallback to numpy conversion which
```
Comment (wconstab) [inline torch/_dynamo/optimizations/backends.py:690]: typo

Comment (shingjan) [reply torch/_dynamo/optimizations/backends.py]: Keen eyes! Fixed.

**Review by wconstab — APPROVED** [review]

**Comment by shingjan** [comment]
Thanks @ezyang @wconstab for reviewing and merging this PR!

**Pull request closed without merging**